docs/vectorizer/api-reference.md

```diff
@@ -148,6 +152,161 @@ in other management functions.
+
+### ai.loading_document
+
+You use `ai.loading_document` to load the data to embed from a file that is referenced in a column of the source table.
+This file path is internally passed to [smart_open](https://github.com/piskvorky/smart_open), so it supports any protocol that smart_open supports. E.g. S3, GCS, etc.
+You just need to ensure the vectorizer worker has the correct credentials to access the file.
+For S3, you can set the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` environment variables.
+
+#### Example usage
+
+```sql
+SELECT ai.create_vectorizer(
+    'my_table'::regclass,
+    loading => ai.loading_document('file_path'),
+    -- other parameters...
+);
+```
+
+#### Parameters
+
+`ai.loading_document` takes the following parameters:
+
+| Name | Type | Default | Required | Description |
+|------|------|---------|----------|-------------|
+| column_name | text | - | ✔ | The name of the column containing the file path. |
+
+#### Returns
+
+A JSON configuration object that you can use in [ai.create_vectorizer](#create-vectorizers).
+
+## Parsing configuration
+
+You use the parsing configuration functions in `pgai` to define the way data is parsed from the source table. This is useful if you want to load data that is not immediately text data and therefore needs to be parsed first.
+
+The parsing functions are:
+
+- [ai.parsing_auto](#aïoparsing_auto)
+- [ai.parsing_none](#aïoparsing_none)
+- [ai.parsing_pymupdf](#aïoparsing_pymupdf)
+
+### ai.parsing_auto
+
+You use `ai.parsing_auto` to automatically parse the data from the source table with a fitting available parser.
+This parser will decide on the fly if it should use `ai.parsing_pymupdf` or `ai.parsing_none`.
+
+#### Example usage
+
+```sql
+SELECT ai.create_vectorizer(
+    'my_table'::regclass,
+    parsing => ai.parsing_auto(),
+    -- other parameters...
+);
+```
+
+#### Parameters
+
+`ai.parsing_auto` takes the following parameters:
+
+| Name | Type | Default | Required | Description |
+|------|------|---------|----------|-------------|
+| None | - | - | - | - |
+
+#### Returns
+
+A JSON configuration object that you can use in [ai.create_vectorizer](#create-vectorizers).
+
+
+### ai.parsing_none
+
+You use `ai.parsing_none` to load the data as is from the source table.
+
+#### Example usage
+
+```sql
+SELECT ai.create_vectorizer(
+    'my_table'::regclass,
+    parsing => ai.parsing_none(),
+    -- other parameters...
+);
+```   
+
+#### Parameters
+
+`ai.parsing_none` takes the following parameters:
+
+| Name | Type | Default | Required | Description |
+|------|------|---------|----------|-------------|
+| None | - | - | - | - |
+
+#### Returns
+
+A JSON configuration object that you can use in [ai.create_vectorizer](#create-vectorizers).
+
+### ai.parsing_pymupdf
+
+You use `ai.parsing_pymupdf` to parse the data from the source table using [pymupdf](https://pymupdf.readthedocs.io/en/latest/).
```

Comment:
As I commented earlier, this "the source table" doesn't sound pretty clear to me.  https://github.com/timescale/pgai/pull/480/files#r1957956533